Add possibility to pass sqlite flag using Home Assistant Addon config.json

goebelmeier opened the pull request:
This PR parses the HA Addon config.json for sqlite_file variable and if set sets the --sqlite flag to the variables value

To make it work, it needs evcc-io/hassio-addon#38

Commit:
Add possibility to pass sqlite flag using
Home Assistant Addon config.json

Hofyyy commented on docker/bin/entrypoint.sh:
```diff
@@ -8,12 +8,19 @@ if [ -f ${HASSIO_OPTIONSFILE} ]; then
     else
-        echo "starting evcc: 'evcc --config ${CONFIG}'"
-        exec evcc --config ${CONFIG}
+        if [ -z "${SQLITE_FILE}" ]; then
```
Why not checking if SQLITE_FILE is valid and then using it? Currently we can not add further options in future. I would suggest. empty use config file. if not empty check against possible options like sql…

andig replied:
Do we have grep installed?

goebelmeier replied:
@Hofyyy elaborate please... do you want me to check if SQLITE_FILE _is not empty_ ? So simply swapping the if-else-clause?

@andig we have, see Line 8, initialisation of CONFIG. But instead of doing grep-cut-sed magic i'm just using two greps with RegEx's

goebelmeier committed:
Update CONFIG initialisation, as the current grep-cut-sed
does not support more than one entry inside options.json
update to pure regex grep

goebelmeier committed:
Fix updated CONFIG initialisation

goebelmeier commented on docker/bin/entrypoint.sh:
```diff
@@ -5,15 +5,22 @@ set -e
 
 if [ -f ${HASSIO_OPTIONSFILE} ]; then
-    CONFIG=$(grep config_file ${HASSIO_OPTIONSFILE}| cut -d ':' -f 2 | sed s/[\"}]//g )
+    CONFIG=$(grep -o '"config_file": "[^"]*' ${HASSIO_OPTIONSFILE} | grep -o '[^"]*$')
```
Hier musste ich das bestehende Konstrukt ändern, da sonst der Start mit mehr als einer Option in /data/options.json failed:
```
# cat /mnt/data/supervisor/addons/data/49686a9f_evcc/options.json
{
  "config_file": "/config/evcc.yaml",
  "sqlite_file": "/data/evcc.db"
}
```
Vorher:
```
# grep config_file /mnt/data/supervisor/addons/data/49686a9f_evcc/options.json  | cut -d ':' -f 2 | sed s/[\"}]//g
 /config/evcc.yaml,
```

Nachher:
```
# grep -o '"config_file": "[^"]*' /mnt/data/supervisor/addons/data/49686a9f_evcc/options.json | grep -o '[^"]*$'
/config/evcc.yaml
```

goebelmeier committed:
Swapped if-else-Statement as suggested by @Hofyyy
